crates/telio-traversal/src/endpoint_providers/stun.rs

```diff
@@ -512,62 +618,62 @@ impl<Wg: WireGuard, E: Backoff> Runtime for State<Wg, E> {
 
         pin!(updated);
 
-        //If no STUN servers are configured -> nothing to do.
-        //NOTE: this will get cancelled on reconfiguration attempt
-        if self.servers.is_empty() {
+        if let (Some(sockets), false) = (self.sockets.as_ref(), self.servers.is_empty()) {
```

Comment:
This is a small style thing, but maybe it would be slightly easier to read if we would reverse the `if` condition:
```rust
        // We must ensure that sockets are created and that we have a populated
        // server list, before we can continue doing anything.
        let sockets = match (self.sockets.as_ref(), self.servers.is_empty()) {
            (Some(sockets), false) => {sockets},
            (_, _) => {
                //If no STUN servers or sockets are configured -> nothing to do.
                //NOTE: this will get cancelled on reconfiguration attempt
                tokio::select! {
                    _ = pending() => {},
                    update = &mut updated => {
                        return update(self).await;
                    }
                }
                return Ok(());
            }
        };
```


And then leave `tokio::select!` untouched? This would remove one indentation level for it. @stalowyjez What are you thoughts about this?